Tests/MongoSwiftTests/CodecTests.swift

```diff
@@ -319,27 +321,6 @@ final class CodecTests: XCTestCase {
     /// Test decoding/encoding to all possible BSON types
     func testBsonValues() throws {
 
-        // decode from fully de-structured 
-        let doc1: Document = [
```

Comment:
this is removed because I changed the `Document.init(from:)` method to always short-circuit if the decoder being used is a `BsonDecoder`. Basically, this is the way that a non-BSON encoder would have encoded a `Document`.

Since this seems like an unlikely use case, I think we should opt for the vast performance improvement of short-circuiting instead.

Reply:
I assume this refers to https://github.com/mongodb/mongo-swift-driver/pull/77/files#diff-cd1c06a2751f196995f7a1679f3b0803R15, but why does that require this test case to be removed?

Is the short-circuit behavior being tested elsewhere?

Reply:
I think this is fine, how often will people be constructing documents this way?

Reply:
yeah, I'm not really sure what the use case would be